### PR TITLE
[webedit] Dynamic delta_time

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About The Pull Request
Makes subsystems that were passing `delta_time` to processed things pass the time since the previous fire.

Webedit because my laptop is out of power and my phone can't run dreammaker. I'll test it later.

## Why It's Good For The Game
It would be nice if lag compensation worked.

## Changelog

:cl: TemporalOroboros
qol: Things are less effected by lag.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
I do not want lag compensation. Delta time exists to make changing subsystem waits possible. I think it's a bad reason to exist, but it's the stated goal. 

I don't want people dropping dead after a ss finally ticks. We fire when we can, no more no less.

On the off chance someone disagrees with me, I'm leaving this pr open for now.

**Comment by TemporalOroboros** [comment]
> I see that reagents has it implemented already. Not sure how I feel about that

Reagents has it implemented incorrectly. It's the time of the previous process is being updated before the process loop so if it runs into `MC_TICK_CHECK` the remaining reactions for that run will run with a greatly reduced delta_time.

**Comment by TemporalOroboros** [comment]
> I do not want lag compensation. Delta time exists to make changing subsystem waits possible. I think it's a bad reason to exist, but it's the stated goal.

The original intent of the PR that added delta_time was to use the time since the last fire for lag compensation. It was changed to it's current state because the PR author was worried about a performance hit.

**Comment by TemporalOroboros** [comment]
Actually, I'm going to make it per datum when I get my laptop charged. If it's too much of a performance hit I can always rever the commit.

File: code/controllers/subsystem/tgui.dm [hunk 1]
```diff
@@ -39,16 +39,23 @@ SUBSYSTEM_DEF(tgui)
...
 	var/list/current_run = src.current_run
 	while(current_run.len)
 		var/datum/tgui/ui = current_run[current_run.len]
 		current_run.len--
 		// TODO: Move user/src_object check to process()
 		if(ui?.user && ui.src_object)
-			ui.process(wait * 0.1)
+			delta_time = min(now - ui.last_processed, max_delta_time) / (1 SECONDS)
+			ui.process(delta_time)
+			ui.last_processed += delta_time
```
Comment (stylemistake) [inline code/controllers/subsystem/tgui.dm:55]: Why does this touch SStgui, its ticks are completely irrelevant, its ticks do not make any effects of the game world.

**Comment by Ryll-Ryll** [comment]
It seems like @LemonInTheDark has already expressed concerns (that I fully share) about how tying processes to realtime as opposed to a player 's ability to respond to their environment (ticks) isn't a good idea, and would just frustrate players. 

Unless you've got a pitch that's way stronger than your response to lemon, I don't think this PR is mergable

**Pull request closed without merging**